src/queries/SearchResultQuery.ts

```diff
@@ -48,6 +66,68 @@ class SearchResultQuery extends BaseQuery<SearchResult> implements SearchResultQ
 
     this.filterQuery['keyword'] = { contains: this.keyword };
   }
+
+  private async buildUrlToFilterQueryParams() {
+    if (!this.url) return;
+
+    this.filterQuery['OR'] = [
+      {
+        adWordTopUrls: {
+          has: this.url,
+        },
+      },
+      {
+        nonAdUrls: {
+          has: this.url,
+        },
+      },
+    ];
+  }
+
+  private async buildUrlWithWordFilterQueryParams() {
+    if (!this.filterAdUrlWord) return;
+
+    const wordToSearch = this.filterAdUrlWord;
+    const searchResults = await this.databaseModal.findMany({ where: { userId: this.userId } });
+    const validIds = searchResults
+      .filter((searchResult: SearchResult) =>
+        searchResult.adWordTopUrls.join(',').includes(wordToSearch)
+      )
+      .map((searchResult: SearchResult) => searchResult.id);
+
+    const existingFilterIds = this.filterQuery.id?.in ?? [];
+    this.filterQuery['id'] = {
+      in: [...existingFilterIds, ...validIds],
+    };
+
+    // TODO: Remove after the process gets approved
+    // return dbClient.$executeRaw`
+    //   SELECT * FROM search_results sr
+    //    WHERE sr."userId" = ${this.userId}
+    //     AND array_to_string(sr.ad_word_top_urls, ', ') ~ ${this.filterAdUrlWord}
+    // `;
+  }
+
+  private async buildUrlContainsAtLeastSpecificCharactersQueryParams() {
+    if (!this.filterUrlCharacter) return;
```

Comment:
```suggestion
    if (!this.filterUrlCharacter) return;
    
```

Let's have a line break